File: python/leetcode/204.py
class Solution(object):
    def countPrimes(self, n):
        if n <= 1:
            return 0
        prime = [1] * n
        prime[0], prime[1] = 0, 0
        i = 2
        while i * i < n:
            prime[i * i: n: i] = [0] * ((n - 1 - i * i) / i + 1)
            i += 1
        return sum(prime)

File: cpp/leetcode/204.cpp
class Solution {
public:
    int countPrimes(int n) {
        vector<bool> prime(n, false);
        int count = 0;
        for (int i = 2; i < n; i++) {
            if (!prime[i]) {
                count++;
                for (int j = 2; i * j < n; j++) {
                    prime[i * j] = true;
                }
            }
        }
        return count;
    }
};